state/core_access_test.go

```diff
@@ -89,6 +92,27 @@ func TestSubmitPayForBlob(t *testing.T) {
 			}
 		})
 	}
+
+	// async blob submission
+	hashCh := make(chan string)
+	for _, tc := range testcases {
+		if tc.asyncSubmission {
+			t.Run(tc.name, func(t *testing.T) {
+				go func() {
+					resp, err := ca.SubmitPayForBlob(ctx, tc.fee, tc.gasLim, tc.blobs)
+					if err == nil {
+						hashCh <- resp.TxHash
+					}
+				}()
+
+				ctx, cancel := context.WithTimeout(ctx, time.Second*3)
+				resp, err := ca.signer.ConfirmTx(ctx, <-hashCh)
+				cancel()
+				require.NoError(t, err)
+				require.EqualValues(t, 0, resp.Code)
+			})
+		}
+	}
```

Comment:
If `ca.SubmitPayForBlob` returns non nill error, the test will hang forever because of deadlock on reading from `<-hashCh` and also no meaningful error will be logged or returned. To fix this lets do:
- use select with `ctx.Done` for reading from `<-hashCh` so test can fail fast
- use `reqiure.NoError` for error produced by `ca.SubmitPayForBlob` to know what happen if it failed.